packages/edit-post/src/store/actions.js

```diff
@@ -478,21 +478,14 @@ export const initializeMetaBoxes =
 		metaBoxesInitialized = true;
 
 		// Save metaboxes on save completion, except for autosaves.
-		addFilter(
-			'editor.__unstableSavePost',
+		addAction(
+			'editor.savePost',
 			'core/edit-post/save-metaboxes',
-			( previous, options ) =>
-				previous.then( () => {
-					if ( options.isAutosave ) {
-						return;
-					}
-
-					if ( ! select.hasMetaBoxes() ) {
-						return;
-					}
-
-					return dispatch.requestMetaBoxUpdates();
-				} )
+			async ( options ) => {
+				if ( ! options.isAutosave && select.hasMetaBoxes() ) {
+					await dispatch.requestMetaBoxUpdates();
```

Comment:
Returning here previously didn't make much sense, and that's why we're removing it, right?

Reply:
Previously we were returning a promise, so that the next filter can `.then` on it. Now this is all handled by the async runner: it will ensure that the next handler runs only after the previous one finished and succeeded.

That's also why this hook is no longer a filter, but an action. Action handlers don't return anything.